app/web_hub.go

```diff
@@ -445,6 +457,7 @@ func (h *Hub) Start() {
 						default:
 							mlog.Error("webhub.broadcast: cannot send, closing websocket for user", mlog.String("user_id", webCon.UserId))
 							close(webCon.Send)
+							webCon.SetClosed()
```

Comment:
Is there still a race condition between the `close` and setting this atomic bit?

Is it strictly necessary to close the channel at all? It appears we already support shutting down the goroutine via a `endWritePump`. If we're not using this as a signal mechanism, can we leave it to garbage collection alone?

Reply:
> Is there still a race condition between the close and setting this atomic bit?

I don't believe there will be. Because this entire thing happens in a single goroutine. Even if the scheduler preempts the goroutine, before it checks for `webCon.IsClosed()`, it needs to go through `SetClosed` first.

> Is it strictly necessary to close the channel at all? 

AFAIU, yes. Because in `writePump`, we check for closing of the channel and send a close message down the websocket. And _after_ that, we close the websocket which closes the `endWritePump`. So it's not just a signal mechanism, we indeed do some cleanup work when the channel is closed.

However, this is my first time working with this part of the codebase, and you obviously have better historical context. So I am open to any improvements that you have in mind.

Reply:
If it is used in one goroutine do you need to use atomic?

Reply:
Strictly speaking, we don't. But this entire thing itself is happening in a separate goroutine. So it makes me a bit more comfortable knowing that it's not going to cause any troubles later, if it gets used in some other place.

Perhaps we can unexport the method, to reduce chances of it being used from other packages. There's no need to export it.

I am not a big fan of the solution in the first place. Totally open to any other solution that you may have in mind :)